Library/Homebrew/override.rb

```diff
@@ -0,0 +1,53 @@
+# typed: true
+# frozen_string_literal: true
+
+module Homebrew
+  # Helper module for overriding machine-specific methods.
+  #
+  # @api private
+  class Override
```

Comment:
```suggestion
  class SystemOverride
```
```suggestion
  class SystemSimulate
```
```suggestion
  class SimulateSystem
```

or something?

Reply:
I kinda like `SimulateSystem`...

The call would be `Homebrew::SimulateSystem.arch = :intel` or `Homebrew::SimulateSystem.os = :monterey`